### PR TITLE
Fixes #37400 - Add setting to control validation of host/lifecycle environment/content source coherence

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

Adds a Setting to allow you to skip running the validation `ensure_valid_content_source`. In certain edge-case situations, this validation can cause problems with host registrations.

#### Considerations taken when implementing this change?

Turning this setting off will erase the guardrails introduced in https://github.com/Katello/katello/pull/10524. Therefore, it defaults to on and the vast majority of users should never need to touch it.

Regardless of the value of the setting, the web UI will still look at coherence of the assigned content source/LCE, and you'll still get UI niceties like those pictured in https://github.com/Katello/katello/pull/10524.

#### What are the testing steps for this pull request?

set up an external smart proxy
Don't sync any environments to it (not even Library)
Register the host to the external smart proxy.

With the setting on, you'll get an error. With the setting off, registration will succeed with no errors.

### CODE REVIEW

File: lib/katello/plugin.rb [hunk 1]
```diff
@@ -512,6 +512,12 @@ def katello_template_setting_values(name)
         description: N_("Default download policy for Smart Proxy syncs (either 'inherit', immediate', or 'on_demand')"),
         collection: proxy_download_policies
 
+      setting 'validate_host_lce_content_source_coherence',
+        type: :boolean,
+        default: true,
+        full_name: N_('Validate host/lifecycle environment/content source coherence'),
+        description: N_("Validate that a host's assigned lifecycle environment is synced by the smart proxy from which the host will get its content")
```
Comment (ianballou) [inline lib/katello/plugin.rb:519]:
```suggestion
        description: N_("Validate that a host's assigned lifecycle environment is synced by the smart proxy from which the host will get its content at registration time")
```

The change content source UI will still block this.

**Review by ianballou — COMMENTED** [review]
Works, with the setting on I can register to smart proxies that don't sync LCEs. Without the setting, at registration time, I see:
```
Validation failed: Content view environment content facets is invalid (HTTP error code 422: Unprocessable Entity)
```

File: lib/katello/plugin.rb (hunk 1 above)
Comment (jeremylenz) [reply lib/katello/plugin.rb:519]: There are still some places in the UI that won't, like the legacy Content Host UI and Content host bulk actions. So it's not just registration time.

**Comment by jeremylenz** [comment]
> Works, with the setting on I can register to smart proxies that don't sync LCEs. Without the setting, at registration time, I see:
> 
> ```
> Validation failed: Content view environment content facets is invalid (HTTP error code 422: Unprocessable Entity)
> ```

Interesting, is that the error that's propagated to subscription-manager? The other day we were actually seeing the validation message itself ("does not sync lifecycle environment xxx") which would be better.

**Comment by ianballou** [comment]
@jeremylenz yeah that's how it popped up when I tried registering to a smart proxy that didn't sync the LCE.

**Comment by jeremylenz** [comment]
Ah yeah, I'm pretty sure it's because that validation moved from content_facet to content_view_environment_content_facet.

File: lib/katello/plugin.rb (hunk 1 above)
Comment (ianballou) [reply lib/katello/plugin.rb:519]: Think it's worth noting that the change content source UI alone won't respect the setting then?

**Review by ianballou — APPROVED** [review]
Works great! I see the right errors in bash after registration and in the logs. Am able to work around the errors with the setting.

**Pull request merged**